devito/interfaces.py

```diff
@@ -694,13 +693,24 @@ def __init__(self, *args, **kwargs):
                             'dimension (save=False). This value will be ignored!'
                             % self.name)
                 time_dim = self.time_order + 1
-                self.indices[0].modulo = time_dim
+                # Get time index
+                _t = [var for var in self.indices if var in (t, time)][0]
```

Comment:
can we assume this is the outermost left dimension or not?

Reply:
For our purpose yes, but I think it is reductive to inforce it, so I left it as a possibility to put the dimensions in any order with a warning stating you need to know what you are doing, as in some case, the order may not matter

Reply:
One seriously big (purely technical) issue is that all those ``if t/time is not indices/dimensions then ...``  make the code a mess